appsec/src/extension/request_lifecycle.c

```diff
@@ -253,6 +272,28 @@ void dd_req_lifecycle_rshutdown(bool ignore_verdict, bool force)
         _do_request_finish_php(ignore_verdict);
         // _rest_globals already called
     }
+
+    // if we don't have service/env, our only chance to update the remote config
+    // path is rshutdown because ddtrace's rinit is called before ours and it
+    // resets the path
+    if (_has_empty_service_or_env() && _rem_cfg_path_changed(false)) {
+        mlog_g(dd_log_debug, "No DD_SERVICE/DD_ENV; trying to sync remote "
+                             "config path on rshutdown");
+        dd_conn *conn =
```

Comment:
If the connection isn't already available, you're effectively initialising a (helper) client which will be stuck waiting for a `client_init`, isn't that the case?

Reply:
right, `dd_helper_mgr_cur_conn` makes more sense here